src/biz/bokhorst/xprivacy/XPrivacy.java

```diff
@@ -29,64 +28,107 @@
 		// Check for LBE security master
 		if (Util.hasLBE()) {
 			Util.log(null, Log.ERROR, "LBE installed");
 			return;
 		}
+		
+		/*
+		 * ActivityManagerService is the beginning of the main "android" process. 
+		 * This is where the core java system is started, where the system context is created and so on. 
+		 * In pre-lollipop we can access this class directly, but in lollipop we have to visit ActivityThread first, 
+		 * since this class is now responsible for creating a class loader that can be used to access ActivityManagerService. 
+		 * It is no longer possible to do so via the normal boot class loader. Doing it like this will create a consistency between older and newer 
+		 * Android versions.
+		 * 
+		 * Note that there is no need to handle arguments in this case. And we don't need them so in case they change over time, we will simply use 
+		 * the hookAll feature. 
+		 */
+		try {
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+				XposedBridge.hookAllMethods(Class.forName("android.app.ActivityThread"), "systemMain", new XC_MethodHook() {
+					@Override
+					protected void afterHookedMethod(MethodHookParam param) throws Throwable {
+						final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+						
+						try {
+							XposedBridge.hookAllConstructors(Class.forName("com.android.server.am.ActivityManagerService", false, loader), new XC_MethodHook() {
+								@Override
+								protected void afterHookedMethod(MethodHookParam param) throws Throwable {
+									bootstrapSystem(param.thisObject, loader);
+								}
+							});
+							
+						} catch (Throwable ex) {
+							Util.bug(null, ex); mHasErr = true;
+						}
+					}
+				});
+				
+			} else {
+				XposedBridge.hookAllMethods(Class.forName("com.android.server.am.ActivityManagerService"), "startRunning", new XC_MethodHook() {
+					@Override
+					protected void afterHookedMethod(MethodHookParam param) throws Throwable {
+						bootstrapSystem(param.thisObject, null);
+					}
+				});
+			}
+			
+			bootstrapZygote();
 
-		init(startupParam.modulePath);
+		} catch (Throwable ex) {
+			Util.bug(null, ex); mHasErr = true;
+		}
 	}
-
+	
 	public void handleLoadPackage(final LoadPackageParam lpparam) throws Throwable {
 		// Check for LBE security master
-		if (Util.hasLBE())
+		int uid = Process.myUid();
+		
+		if (Util.hasLBE() || mHasErr || uid <= Process.SYSTEM_UID)
 			return;
 
-		handleLoadPackage(lpparam.packageName, lpparam.processName, lpparam.classLoader, lpparam.isFirstApplication,
-				mSecret);
+		bootstrapPackage(lpparam.packageName, lpparam.classLoader);
 	}
+	
 
-	// Common
-	private static void init(String path) {
-		Util.log(null, Log.WARN, "Init path=" + path);
-
+	private void bootstrapZygote() throws Throwable {
 		// Generate secret
 		mSecret = Long.toHexString(new Random().nextLong());
+		
+		/*
+		 * Zygote can access this file (Even with SELinux enabled) as it is running as the outer root. 
```

Comment:
Are you very sure about /data/system be accessible by initZygote?
Not every ROM is AOSP and some manufacturers have very tight SELinux policies.
As said before, this can result in bootloops.